velox/vector/tests/VectorTest.cpp

```diff
@@ -3933,5 +3933,49 @@ TEST_F(VectorTest, hasOverlappingRanges) {
   test(2, {false, false}, {1, 0}, {1, 3}, true);
 }
 
+TEST_F(VectorTest, sliceBigintBuffer) {
+  auto bufferPtr = AlignedBuffer::allocate<int64_t>(10, pool());
+  auto sliceBufferPtr =
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, 1, 5, pool());
+  ASSERT_TRUE(sliceBufferPtr->isView());
+  ASSERT_EQ(sliceBufferPtr->size(), 40); // 5 * type size of int64_t.
+  ASSERT_EQ(sliceBufferPtr->as<int64_t>(), bufferPtr->as<int64_t>() + 1);
+
+  VELOX_ASSERT_USER_THROW(
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, -1, 5, pool()),
+      "Offset must be non-negative.");
+  VELOX_ASSERT_USER_THROW(
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, 0, -1, pool()),
+      "Length must be non-negative.");
+  VELOX_ASSERT_USER_THROW(
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, 11, 1, pool()),
+      "Offset must be less than or equal to 10.");
+  VELOX_ASSERT_USER_THROW(
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, 5, 6, pool()),
+      "Length must be less than or equal to 5.");
+}
+
+TEST_F(VectorTest, sliceBooleanBuffer) {
```

Comment:
Would you add a test for slicing ARRAY buffer (should fail)?

Reply:
Yes it triggers a failure, test is added, please help check.